src/error.ts

```diff
@@ -15,11 +15,47 @@ export const isResolutionException = (err: unknown) => {
   return isResolutionEx(JSON.parse(err.message))
 }
 
-export const errorHandler = (err: unknown) => {
-  if (!(err instanceof Error)) return error(`Unexpected error: ${err}`)
+export const isModelingException = (input: unknown): input is ModelingException => {
+  if (typeof input === 'object' && input !== null && 'ExceptionKind' in input) {
+    const exception = input as ModelingException
+    return isParsingEx(exception) || isResolutionEx(exception)
+  }
+  return false
+}
+
+export const errorHandler = (err: unknown): ModelingException => {
+  try {
+    if (err instanceof Error) {
+      const exception = JSON.parse(err.message) as ModelingException
 
-  const exception = JSON.parse(err.message) as ModelingException
+      if (isParsingEx(exception) || isResolutionEx(exception)) {
+        return exception
+      }
+    }
 
-  if (!(isParsingEx(exception) || isResolutionEx(exception))) error('Unknown exception type')
-  error(exception)
+    return {
+      ExceptionKind: 'Parsing',
+      Errors: [
+        {
+          Cause: 'An unknown error occurred.',
+          Action: 'Check the input or contact support.',
+          ValidationID: 'unknown:error',
+          Value: JSON.stringify(err),
+        },
+      ],
+    } as const
+  } catch (parseErr) {
+    error('Failed to parse error message:', parseErr)
+    return {
+      ExceptionKind: 'Parsing',
+      Errors: [
+        {
+          Cause: 'Failed to parse error message.',
+          Action: 'Ensure the parser returns valid ModelingException JSON.',
+          ValidationID: 'error:unparseableException',
+          Value: JSON.stringify(err),
+        },
+      ],
+    } as const
```

Comment:
if you wanted to simplify things, it doesn't seem like there's much difference in these two errors to me, since an error with an unparsable message will always be an unknown error

Reply:
also not sure that unknown errors should be of `ModelingException` type, since they happen when an error _isn't_ a `ModelingException`

Reply:
could maybe just rethrow if the error isn't a `ModelingException`...